### PR TITLE
REF/TYP: use OpsMixin for DataFrame

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

made possible by #36483

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -420,6 +421,7 @@ def _constructor(self) -> Type[DataFrame]:
     _constructor_sliced: Type[Series] = Series
     _deprecations: FrozenSet[str] = NDFrame._deprecations | frozenset([])
     _accessors: Set[str] = {"sparse"}
+    __hash__ = NDFrame.__hash__
```
Comment (jreback) [inline pandas/core/frame.py:424]: was this defined before?

**Review by jreback — CHANGES_REQUESTED** [review]
also checks are failing

File: pandas/core/frame.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/frame.py:424]: ATM this PR puts OpsMixin ahead of NDFrame in the MRO, which is why this needs to be re-pinned.  flipping those should make this unnecessary

**Comment by jreback** [comment]
wouldn't hurt to merge master

**Comment by jbrockmendel** [comment]
updated+green

**Review by jreback — APPROVED** [review]

**Pull request merged**